conans/test/integration/lockfile/test_user_overrides.py

```diff
@@ -102,3 +105,73 @@ def test_user_python_overrides():
     # The resulting lockfile contains the full revision now
     new_lock = c.load("new.lock")
     assert "pytool/1.1" in new_lock
+
+
+def test_add_revisions():
+    """ Is it possible to add revisions explicitly too
+    """
+    c = TestClient()
+    c.save({"math/conanfile.py": GenConanfile("math"),
+            "engine/conanfile.py": GenConanfile("engine", "1.0").with_requires("math/[*]"),
+            "game/conanfile.py": GenConanfile("game", "1.0").with_requires("engine/[*]")})
+
+    c.run("export math --version=1.0")
+    rev0 = c.exported_recipe_revision()
+    c.save({"math/conanfile.py": GenConanfile("math").with_build_msg("New rev1")})
+    c.run("export math --version=1.0")
+    rev1 = c.exported_recipe_revision()
+    c.save({"math/conanfile.py": GenConanfile("math").with_build_msg("New rev2")})
+    c.run("export math --version=1.0")
+    rev2 = c.exported_recipe_revision()
+
+    c.run("export engine")
+    c.run("graph info game")
+    assert f"math/1.0#{rev2}" in c.out
+    assert f"math/1.0#{rev1}" not in c.out
+
+    # without revision, it will resolve to latest
+    c.run("lock add --requires=math/1.0 --requires=unrelated/2.0")
+    c.run("graph info game --lockfile=conan.lock --lockfile-out=new.lock --lockfile-partial")
+    assert f"math/1.0#{rev2}" in c.out
+    assert f"math/1.0#{rev1}" not in c.out
+    assert f"math/1.0#{rev0}" not in c.out
+    # The resulting lockfile contains the full revision now
+    new_lock = c.load("new.lock")
+    assert f"math/1.0#{rev2}" in new_lock
+    assert f"math/1.0#{rev1}" not in new_lock
+    assert f"math/1.0#{rev0}" not in c.out
+
+    # with revision, it will resolve to that revision
+    c.run(f"lock add --requires=math/1.0#{rev1} --requires=unrelated/2.0")
+    c.run("graph info game --lockfile=conan.lock --lockfile-out=new.lock --lockfile-partial")
+    assert f"math/1.0#{rev1}" in c.out
+    assert f"math/1.0#{rev2}" not in c.out
+    assert f"math/1.0#{rev0}" not in c.out
+    # The resulting lockfile contains the full revision now
+    new_lock = c.load("new.lock")
+    assert f"math/1.0#{rev1}" in new_lock
+    assert f"math/1.0#{rev2}" not in new_lock
+    assert f"math/1.0#{rev0}" not in c.out
```

Comment:
I don't get why this test is related to the change. The change is:

```python
        # When we are adding something manually to a lockfile, we want to prioritize it
        # so we make it the first element in the list
        self._requires.move_to_end(ref, last=False)
```
But, here, we're not checking any order of the requirements, right? This test is testing only that we're able to add manually revisions, isn't it?

Reply:
The other test is checking the actual order in the lockfile. This test is validating that adding multiple requirements to the lockfile, even if incomplete, it won't crash (it did crash before, because of the order was raising TypeError comparison between float <> None or str <> None